### ISSUE TITLE
links in twisted core docs are broken

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4952]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)| @habnabit reported|
|-|-|
|Trac ID|trac#4952|
|Type|defect|
|Created|2011-03-15 05:37:37Z|

Spidered the most recent docs; all of these links are broken in the documentation. Format is the document with broken links, followed by the broken links.

```
http://twistedmatrix.com/documents/current/core/development/policy/doc-standard.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.web.widgets.Gadget.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.web.widgets.Widget.html

http://twistedmatrix.com/documents/current/core/howto/application.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.internet.interfaces.IReactorUDP.connectUNIXDatagram.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.internet.interfaces.IReactorUDP.listenMulticast.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.internet.interfaces.IReactorUDP.listenUNIXDatagram.html

http://twistedmatrix.com/documents/current/core/howto/basics.html
  http://twistedmatrix.com/documents/10.2.0/api/distutils.html

http://twistedmatrix.com/documents/current/core/howto/cred.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.cred.checkers.ICheckerFactory.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.protocols.pop3.IMailbox.html

http://twistedmatrix.com/documents/current/core/howto/glossary.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.cred.checkers.ICredentialChecker.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.im.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.protocols.sux.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.python.components.Interface.html

http://twistedmatrix.com/documents/current/core/howto/logging.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.python.log.msg.html

http://twistedmatrix.com/documents/current/core/howto/pb-copyable.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.pb.Cacheable.getStateToCacheAndObserveFor.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.pb.Cacheable.stoppedObserving.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.pb.Copyable.getStateToCopy.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.pb.RemoteCopy.setCopyableState.html

http://twistedmatrix.com/documents/current/core/howto/pb-cred.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.cred.checkers.%0A.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.cred.checkers.ANONYMOUS.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.flavors.Viewable.html

http://twistedmatrix.com/documents/current/core/howto/pb-usage.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.Referenceable.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.pb.InsecureJelly.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.spread.pb.Jelly.html

http://twistedmatrix.com/documents/current/core/howto/process.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.internet.interfaces.%0A.html

http://twistedmatrix.com/documents/current/core/howto/producers.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.internet.interfaces.IPullProducer.stopProducing.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.internet.interfaces.IPushProducer.stopProducing.html

http://twistedmatrix.com/documents/current/core/howto/testing.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.trial.unittest.TestCase.assertWarns.html

http://twistedmatrix.com/documents/current/core/howto/upgrading.html
  http://twistedmatrix.com/documents/10.2.0/api/Versioned.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.internet.app.Application.html
```

Attachments:

* [doc-standard-4952.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/99f/99f3b3ada2b4675c518ff23cbd9539da05e2f1f8/ecfc96277b56a90d4066fe2895e82a550710f7e4.patch) (1991 bytes) - added by thijs on 2011-06-27 16:59:09Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4952 4952
type__defect defect
reporter__habnabit habnabit
priority__normal normal
milestone__ 
branch__ 
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1300167457000000 1300167457000000
changetime__1309362996000000 1309362996000000
version__None None
owner__thijs thijs
cc__thijs cc__dustin
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)<a name="note_1"></a>|@habnabit commented|
|-|-|

Oh, also one API document is broken

```
http://twistedmatrix.com/documents/10.2.0/api/twisted.web.client.html
  http://twistedmatrix.com/documents/10.2.0/api/twisted.web.client.WebClientContextFactory%200.html
```

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Replying to [habnabit](#4952):
> http://twistedmatrix.com/documents/current/core/development/policy/doc-standard.html
>   http://twistedmatrix.com/documents/10.2.0/api/twisted.web.widgets.Gadget.html
>   http://twistedmatrix.com/documents/10.2.0/api/twisted.web.widgets.Widget.html

This was removed with woven (r25924) so we need a new example here.

> 
> http://twistedmatrix.com/documents/current/core/howto/application.html

The current version of the howto doesn't contain broken links according to the [link checker](http://validator.w3.org/checklink?uri=http%3A%2F%2Ftwistedmatrix.com%2Fdocuments%2Fcurrent%2Fcore%2Fhowto%2Fapplication.html&hide_type=all&depth=&check=Check).

> 
> http://twistedmatrix.com/documents/current/core/howto/basics.html

Fixed in r31812.

> 
> http://twistedmatrix.com/documents/current/core/howto/cred.html

Fixed in r31812.

> 
> http://twistedmatrix.com/documents/current/core/howto/glossary.html

Link checker found 3 broken links in the current glossary which were fixed in r31812.

> 
> http://twistedmatrix.com/documents/current/core/howto/logging.html

All links look [valid](http://validator.w3.org/checklink?uri=http%3A%2F%2Ftwistedmatrix.com%2Fdocuments%2Fcurrent%2Fcore%2Fhowto%2Flogging.html&hide_type=all&depth=&check=Check).

> 
> http://twistedmatrix.com/documents/current/core/howto/pb-copyable.html

Fixed in r31014.

> 
> http://twistedmatrix.com/documents/current/core/howto/pb-cred.html

Fixed in r31014.

> 
> http://twistedmatrix.com/documents/current/core/howto/pb-usage.html

Fixed in r31014.

> 
> http://twistedmatrix.com/documents/current/core/howto/process.html

Fixed  in r31812.

> 
> http://twistedmatrix.com/documents/current/core/howto/producers.html

Fixed in r30996.

> 
> http://twistedmatrix.com/documents/current/core/howto/testing.html

Fixed in r31812.

> 
> http://twistedmatrix.com/documents/current/core/howto/upgrading.html

Fixed in r31099.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [thijs](#note_3):
> Replying to [habnabit](#4952):
> > http://twistedmatrix.com/documents/current/core/development/policy/doc-standard.html
> >   http://twistedmatrix.com/documents/10.2.0/api/twisted.web.widgets.Gadget.html
> >   http://twistedmatrix.com/documents/10.2.0/api/twisted.web.widgets.Widget.html
> 
> This was removed with woven (r25924) so we need a new example here.

Attached patch updates the example with `Resource.putChild`. (the `twisted/topfiles/4952.misc` file somehow doesn't show up in the trac but its in the patch)

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_5"></a>|@dustin set owner to @thijstriemstra|
|-|-|

The patch looks good.  It's simple enough - the content of the links here are not particularly critical anyway, but they *do* exist.

It doesn't look like there are any contents to ```4952.misc```:
```
Index: twisted/topfiles/4952.misc
===================================================================
Index: doc/core/development/policy/doc-standard.xhtml
```
which is probably why it's not showing up in Trac.  That will need to be included in the patch before it can be committed.

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_6"></a>|@dustin commented|
|-|-|

Hah, I'm a dummy:
 * misc - for tickets which are very minor and not worth summarizing outside of the svn changelog. These should be empty (their contents will be ignored). 
(from [ReviewProcess](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles)).  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [32198]) Apply doc-standard-4952.patch: Correct some broken example API links in the doc-standard policy.

Author: thijs
Reviewer: dustin
Fixes: [#4952](https://github.com/twisted/twisted/issues/4952)
